code/build-analysis/03b_county-census.R

```diff
@@ -0,0 +1,219 @@
+          "P12_029N",
+          "P12_030N",
+          "P12_044N", # total female 65+
+          "P12_045N",
+          "P12_046N",
+          "P12_047N",
+          "P12_048N",
+          "P12_049N")
+
+
+
+## load census data using get_decennial function from tidycensus
+cen <- get_decennial(year = 2020,
+                     geography = "county",
+                     keep_geo_vars = TRUE,
+
+                     variables = vars,
+                     sumfile = "dhc",
+                     output = "wide")  |>
+  dplyr::rename(tot = P1_001N,
+         tot_urb_rur = P2_001N,
+         tot_urb = P2_002N,
+         tot_own = H4_001N,
+         tot_own_mort = H4_002N,
+         tot_own_free = H4_003N,
+         tot_age = P12_001N,
+         tot_races = P3_001N,
+         white = P3_002N,
+         black = P3_003N,
+         hisp_og_tot = P4_001N,
+         hisp_og = P4_003N,
+         hisp_race_tot = P5_001N,
+         his_race = P5_010N) |>
+  mutate(prop_urb = tot_urb/tot_urb_rur * 100, #calculate proportion vars
+         prop_white = white/tot_races * 100,
+         prop_black = black/tot_races * 100,
+         prop_hisp = hisp_og/hisp_og_tot * 100,
+         prop_hisp_race = his_race/hisp_race_tot * 100,
+         own_occupy = rowSums(across(c(tot_own_mort, tot_own_free)),
+                                   na.rm = TRUE),
+         prop_own_occ = own_occupy/tot * 100,
+         all18 = rowSums(across(c(P12_003N,
+                                   P12_004N,
+                                   P12_005N,
+                                   P12_006N,
+                                  P12_027N, P12_028N,
+                                  P12_029N, P12_030N)),
+                          na.rm = TRUE),
+         all65 = rowSums(across(c(P12_020N, P12_021N,
+                                   P12_022N, P12_023N,
+                                   P12_024N, P12_025N,
+                                  P12_044N, P12_045N,
+                                  P12_046N, P12_047N,
+                                  P12_048N, P12_049N)),
+                          na.rm = TRUE),
+         prop_u18 = all18/tot_age * 100,
+         prop_o65 = all65/tot_age * 100,
+         county = str_trim(str_extract(NAME,
+                                       ".*(?= County)")),  # extract everything before "County"
+         state = str_trim(str_extract(NAME,
+                                      "(?<=, ).*")), # extract everything after common and space
+         county = toupper(county)) |>
+  select(-starts_with("P1"))
+
+sum_vars <- c("prop_urb", "prop_white", # designate our final summary variables
+              "prop_black", "prop_hisp",
+              "prop_u18", "prop_o65",
+              "prop_own_occ")
+
+##calc summary stats for all counties
+summary_all <- bind_rows(
+  lapply(sum_vars,
+         function(var) calc_summary_stats(cen,
+                                          var))
+)
+
+
+
+
+
+##load cvr data subset
+cty_codes <- fips_codes |>
```

Comment:
rather than using the .csv file that is in the combined folder in the parquet dropbox, using the built-in fips_code table from tidycensus. Our public repo on dataverse doesn't have the combined files and this is easier than making the csv available where this gets posted.

Reply:
Good move, but need to be careful to check if minor spelling discrepancies cause a mismatch. It seems like "ST JOHNS" vs. "ST. JOHNS" was one, but I fixed that